controllers/helmrepository_controller.go

```diff
@@ -605,3 +610,29 @@ func (r *HelmRepositoryReconciler) eventLogf(ctx context.Context, obj runtime.Ob
 	}
 	r.Eventf(obj, eventType, reason, msg)
 }
+
+// reconcileTypeChanged handles the case where the object's type has changed.
+// It will remove the Artifact from the Storage and clean up the object's status.
+// It returns error event if the artifact could not be removed.
+func (r *HelmRepositoryReconciler) reconcileTypeChange(ctx context.Context, obj *sourcev1.HelmRepository) (sreconcile.Result, error) {
```

Comment:
Can't this just be wired into `reconcileDelete`? I do not feel a type change is much different from it, except for it not removing the finalizer which can be determined by the fact that the deletion timestamp is not set.